File: golang/57-insert-interval.go
package golang

import "math"

func insert(intervals [][]int, newInterval []int) [][]int {
	var result [][]int
	for i := 0; i < len(intervals); i++ {
		if newInterval[0] > intervals[i][1] {
			result = append(result, intervals[i])
			continue
		}
		if newInterval[1] < intervals[i][0] {
			result = append(result, newInterval)
			result = append(result, intervals[i:]...)
			return result
		}
		newInterval[0] = int(math.Min(float64(newInterval[0]), float64(intervals[i][0])))
		newInterval[1] = int(math.Max(float64(newInterval[1]), float64(intervals[i][1])))
	}
	result = append(result, newInterval)
	return result
}

File: javascript/57-insert-interval.js
/**
 * @param {number[][]} intervals
 * @param {number[]} newInterval
 * @return {number[][]}
 */
 var insert = function(intervals, newInterval) {    
    const result = []
    for (let i = 0; i < intervals.length; i++) {
        if (newInterval[0] > intervals[i][1]) {
            result.push(intervals[i])
        } else if (newInterval[1] < intervals[i][0]) {
            result.push(newInterval)
            return result.concat(intervals.slice(i))
        } else {
            newInterval = [Math.min(newInterval[0], intervals[i][0]), Math.max(newInterval[1], intervals[i][1])]
        }
    }
    result.push(newInterval)
    return result
};